cpc/include/cpc_sketch_impl.hpp

```diff
@@ -401,7 +406,7 @@ string<A> cpc_sketch_alloc<A>::to_string() const {
 
 template<typename A>
 void cpc_sketch_alloc<A>::serialize(std::ostream& os) const {
-  compressed_state<A> compressed;
+  compressed_state<A> compressed(A(sliding_window.get_allocator()));
```

Comment:
Should serialize() have the option of using a different allocator? It's not entirely obvious that, say, a limited-size allocator would want transient things like serialization to complete for space with the main stored data?

Reply:
We are using a provided allocator for all allocations within an instance, including temporary allocations. I think this is good enough.